src/corehost/corehost.cpp

```diff
@@ -295,10 +303,20 @@ int get_com_activation_delegate(pal::string_t *app_path, com_activation_fn *dele
 
     // Leak fxr
 
-    auto entry_del = (hostfxr_get_com_activation_delegate_fn)pal::get_symbol(fxr, "hostfxr_get_com_activation_delegate");
-    if (entry_del == nullptr)
+    auto get_com_delegate = (hostfxr_get_com_activation_delegate_fn)pal::get_symbol(fxr, "hostfxr_get_com_activation_delegate");
+    if (get_com_delegate == nullptr)
         return StatusCode::CoreHostEntryPointFailure;
 
+    auto set_error_writer = (hostfxr_set_error_writer_fn)pal::get_symbol(fxr, "hostfxr_set_error_writer");
+    if (set_error_writer != nullptr)
+    {
+        // Set the error writer during COM activation to the default.
+        // If the previous value is non-null, then set it to the defined value.
+        hostfxr_error_writer_fn prev = set_error_writer(libhost_default_error_writer);
+        if (prev != nullptr)
+            (void)set_error_writer(prev);
```

Comment:
I don't understand this - the end result is that if there was a writer, you leave it there.
* There really should not be one - otherwise whoever set it made a mistake
* Regardless - we don't want to write the COM related errors to a random writer - think about it as operation-scoped. COM operation should be isolated from everything else going on, just like you use ALC to isolate its assembly loading.

Reply:
I disagree here. Since this is thread local, if the thread was used to activate a CoreCLR instance, then it should be respected. If it wasn't set, then we keep the value, but users should have the final say if they set an error writer no?

Reply:
The scenario I'm thinking about is like this:
* Host (for example our apphost in GUI mode would do this) which sets a custom error writer
* It calls into hostfxr to execute an app
* Later on the app does a COM activation on an object which happens to be written in .NET Core as well - it could be that it's trying things out and fully expects the operation to fail.
* The code flow gets into this code (as part of the COM activation) - given the code we would leave the custom error writer in place
* The COM activation fails - we write some errors to the custom writer - so in our case it goes all the way down to the original apphost
* The managed app catches the exception/failure and moves on since it expected the failure
* Much later on it exits - code gets to the apphost which sees that there are some unreported errors - so it pops up a dialog with the error message - which makes no sense in that context.

In general I view the COM activation as a self-contained operation which should not leak anything outside - errors included. Or maybe a different way to put it - the caller should be able to handle the error completely, which this wouldn't allow.

Reply:
> * Much later on it exits - code gets to the apphost which sees that there are some unreported errors - so it pops up a dialog with the error message - which makes no sense in that context.

@vitek-karas I'm sorry but this is test nonsense and just back practice. The scenario described is around how poorly this repo does testing _not_ helping a developer understand what is going on in their application. By explicitly setting the activation path to the null stream there is no way for a COM server consumer to know why the activation failed. This is making the situation harder. I agree that having a noisy logs is painful but that is more an artifact of just logging everything instead of logging things are aren't expected (i.e. warnings) or actions that fail (i.e. errors). When an error occurs there should be a full dump, but in the case where nothing fails or defaults are expected then the logs should be generally thin. I would be more inclined to have a conversation about how to reduce existing noise in the logs rather than hide COM activation logic which has been and will be with this suggestion the hardest area to debug.

Reply:
Chatted offline with Vitek. I will remove this and instead rely on `IErrorInfo` since that is the COM way.